core/src/consensus.rs

```diff
@@ -404,9 +486,26 @@ impl Tower {
         total_stake: u64,
         epoch_vote_accounts: &HashMap<Pubkey, (u64, Account)>,
     ) -> SwitchForkDecision {
+        let mut stray_restored_ancestors = HashSet::default();
+
         self.last_voted_slot()
             .map(|last_voted_slot| {
-                let last_vote_ancestors = ancestors.get(&last_voted_slot).unwrap();
+                let last_vote_ancestors = if !self.use_stray_restored_ancestors(ancestors, last_voted_slot) {
+                    ancestors.get(&last_voted_slot).unwrap()
```

Comment:
Relaxing an invariant of bank_forks by removing `unwrap()` here might hide other bad conditions. So, I retained `.unwrap()` for the general code path as opposed to original diff: https://github.com/solana-labs/solana/pull/9902/files#diff-d848cba681535238849e7e33ed32702bL353